app/lib/frontend/handlers.dart

```diff
@@ -34,6 +36,14 @@ Future<shelf.Response> appHandler(
 ) async {
   _logPubHeaders(request);
 
+  // do pub.dartlang.org-only routes
+  if (request.requestedUri.host == legacyHost) {
+    final rs = await PubDartlangOrgService().router.handler(request);
```

Comment:
We should not be creating a `router` for every request... That's expensive...

Reply:
I just saw we have similar problem with `PubSiteService(shelfPubApi).router.handler`

Reply:
How about the first handler in `PubSiteService` be something like:
```dart
class PubSiteService {

  final _pubDartlangOrgHandler = PubDartlangOrgService().router.handler;

  @Route.all('/<path|.*>')
  Future<Response> _handleLegacyHost(Request request) {
    if (request.uri.host != legacyHost) {
      return null;
    }
    return _pubDartlangOrgHandler(request);
  }
}
```

Reply:
This is just illustrative code... Maybe we don't have host available?

Reply:
I'd do this in a follow-up, to include other host redirects (e.g. dartdocs.org)